atc/creds/vault/api_client.go

```diff
@@ -196,6 +200,73 @@ func (ac *APIClient) baseClient() (*vaultapi.Client, error) {
 	return client, nil
 }
 
+func (ac *APIClient) configureTLS(config *tls.Config) error {
+	if ac.tlsConfig.CACert != "" || ac.tlsConfig.CACertFile != "" || ac.tlsConfig.CAPath != "" {
+		rootConfig := &rootcerts.Config{
+			CAFile:        ac.tlsConfig.CACertFile,
+			CAPath:        ac.tlsConfig.CAPath,
+			CACertificate: []byte(ac.tlsConfig.CACert),
+		}
```

Comment:
Should we mention somewhere, maybe in the docs, about the precedence of these fields? according to https://github.com/hashicorp/go-rootcerts/blob/master/rootcerts.go#L17-L28, `CAFile` has the highest precedence, followed by `CACertificate`, followed by `CAPath`.

Reply:
I think we should actually just remove support for `CAPath` as we migrate to `var_sources` exclusive config, and at that point we'll only have to support `ca_cert`.